tests/MongoDB.Driver.Core.Tests/Core/Configuration/ConnectionStringTests.cs

```diff
@@ -514,6 +514,59 @@ public void When_a_database_name_is_specified(string connectionString, string db
             subject.DatabaseName.Should().Be(db);
         }
 
+        [Theory]
+        [InlineData("mongodb://localhost/?directConnection=true&replicaSet=yeah", true, ClusterConnectionMode.Direct)]
+        [InlineData("mongodb://localhost/?directConnection=true", true, ClusterConnectionMode.Direct)]
+        [InlineData("mongodb://localhost/?directConnection=false&replicaSet=yeah", false, ClusterConnectionMode.ReplicaSet)]
+        [InlineData("mongodb://localhost/?directConnection=false", false, ClusterConnectionMode.Automatic)]
+        public void When_a_directConenction_is_specified(string connectionString, bool directConnection, ClusterConnectionMode mode)
+        {
+            var subject = new ConnectionString(connectionString);
+
+            var result = bool.Parse(subject.GetOption("directConnection"));
+            result.Should().Be(directConnection);
```

Comment:
Change assertions to:

    subject.Connect.Should().Be(connect);
    subject._directConnection().Should().Be(directConnection);

Note: requires adding a `ConnectionStringReflector` class (unless you add a `DirectConnection` public property).

    public static class ConnectionStringReflector
    {
        public static bool _directConnection(this ConnectionString obj) => (bool)Reflector.GetFieldValue(obj, nameof(_directConnection));
    }